[10.0][FIX] Hugely degraded performance for the translation subquery inject…

eantones opened the pull request:
…ion in _generate_translated_field

It solves issue https://github.com/odoo/odoo/issues/18679 on Odoo v10

Compatible with all PostgreSQL versions supported by Odoo 10.0

eantones committed:
[FIX] Hugely degraded performance for the translation subquery injection in _generate_translated_field #18679

It solves issue #18679 (https://github.com/odoo/odoo/issues/18679) on Odoo v10

Only works on PostgreSQL 9.5 or above

FFernandez-PlanetaTIC committed:
[FIX] base: create index as unique and remove duplicates before

pedrobaeza commented:
You can't do this because PG versions for 10.0 include <= 9.5.

FFernandez-PlanetaTIC commented:
Hi @pedrobaeza,

This fixes a huge performace issue in v10 and v11.

Do you think it's worth making the PR compatible for a smaller PG version?

Thanks!

pedrobaeza commented:
Totally yes. This is not something about making it compatible, but someone using it with that version to not break its compatibility. You should migrate to upper versions instead.

eantones committed:
[FIX] base: Make it compatible with all versions of PostgreSQL supported by Odoo 10.0

eantones commented:
@pedrobaeza you're right, its important to not break the compatibility.

I just added a commit to solve this.

Thanks!

eantones committed:
[FIX] base: Fix issues importing translations

Based on https://github.com/PlanetaTIC/odoo/commit/3b3c9c687d013200db49484d2fbb4a8b45f7a52d correction.

- Previous commit breaks import (from ONLY needed in where not exists)
- Avoid duplicates in import (a slightly clearer and cleaner approach)

eantones commented:
Hi @FFernandez-PlanetaTIC,
    
I corrected it based on your [solution](https://github.com/PlanetaTIC/odoo/commit/3b3c9c687d013200db49484d2fbb4a8b45f7a52d ) but with minor changes.

Thank you!

FFernandez-PlanetaTIC reviewed (approved):
LGTM, thanks!

dek-odoo commented:
Is there a PR for v14 on this issue ?
It seems v14 also has a similar issue..
@eantones
